pyxtrlock/__init__.py

```diff
@@ -1,8 +1,40 @@
 
+import os
+import socket
 import sys
 
+from xdg.BaseDirectory import get_runtime_dir
+
 
 def panic(*message_parts, exit_code=1):
     """Print an error message to stderr and exit"""
     print("pyxtrlock:", *message_parts, file=sys.stderr)
     sys.exit(exit_code)
+
+
+def detect_wayland_session():
+    """
+    Detect whether we're running in a Wayland session and abort if so.
+    """
+
+    if os.environ.get("XDG_SESSION_TYPE") == "x11":
+        return
+
+    if os.environ.get("WAYLAND_DISPLAY"):
+        panic(
+            "WAYLAND_DISPLAY is set, suspecting Wayland session. "
+            "Using pyxtrlock in a Wayland session is insecure. Aborting."
+        )
+
+    xdg_runtime_dir = get_runtime_dir(strict=True)
+    wayland_socket = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM, 0)
+
+    try:
+        wayland_socket.connect(os.path.join(xdg_runtime_dir, "wayland-0"))
+    except OSError:
```

Comment:
Why do we get an OSError, if we successfully connect to the wayland socket. This code panics on X11, when there is no wayland socket.